crates/uv/src/commands/tool/run.rs

```diff
@@ -104,6 +105,26 @@ pub(crate) async fn run(
         return Err(anyhow::anyhow!("Tool command could not be parsed as UTF-8 string. Use `--from` to specify the package name."));
     };
 
+    let possible_path = Path::new(target);
+    if possible_path
+        .extension()
+        .is_some_and(|ext| ext.eq_ignore_ascii_case("py"))
+    {
+        if possible_path.try_exists()? {
+            return Err(anyhow::anyhow!(
+                "It looks like you are trying to run a script. Try `{}{}`",
+                "uv run ".cyan(),
+                possible_path.to_string_lossy().to_string().cyan(),
+            ));
+        }
+        return Err(anyhow::anyhow!(
+            "It looks like you are trying to run a script that doesn't exist. \n\n{}{} Did you mean `uvx {}`?",
```

Comment:
It seems nice to include the script name, maybe like

> It looks like you're trying to run a script (`{}`) but it does not exist.

Not in love with that, but don't have a better idea.

Reply:
In the "Did you mean" message, should we clarify what `uvx <name>` would do? Like "Did you mean to run a tool with `uvx {}`"?

I'm not sure if I prefer it, but wanted to put it out there.

Reply:
Yeah, I think adding that context is helpful